Module 3/SQL_Assignment.txt

```diff
@@ -0,0 +1,1880 @@
+4 rows in set (0.01 sec)
+
+
+2) List all customers with a rating of 100. 
+
+mysql> SELECT cname FROM customers WHERE rating = 100;
++----------+
+| cname    |
++----------+
+| Hoffman  |
+| Clemens  |
+| Pererira |
++----------+
+3 rows in set (0.00 sec)
+
+
+3) Find the largest order taken by each salesperson on each date. 
+
+mysql> SELECT s.snum, s.sname, o.odate
+    -> FROM salespeople AS s
+    -> JOIN orders AS o ON o.snum = s.snum
+    -> GROUP BY o.snum, o.odate;
++------+---------+----------+
+| snum | sname   | odate    |
++------+---------+----------+
+| 1007 | Rifkin  | 10/03/90 |
+| 1004 | Motika  | 10/03/90 |
+| 1001 | Peel    | 10/03/90 |
+| 1002 | Serres  | 10/03/90 |
+| 1002 | Serres  | 10/04/90 |
+| 1001 | Peel    | 10/05/90 |
+| 1003 | AxelRod | 10/04/90 |
+| 1002 | Serres  | 10/06/90 |
+| 1001 | Peel    | 10/06/90 |
++------+---------+----------+
+9 rows in set (0.04 sec)
+
+
+4) Arrange the Orders table by descending customer number. 
+
+mysql> SELECT * FROM orders ORDER BY cnum;
++------+---------+----------+------+------+
+| onum | amt     | odate    | cnum | snum |
++------+---------+----------+------+------+
+| 3003 |  767.19 | 10/03/90 | 2001 | 1001 |
+| 3009 | 1713.23 | 10/04/90 | 2002 | 1003 |
+| 3005 | 5160.45 | 10/03/90 | 2003 | 1002 |
+| 3007 |   75.75 | 10/04/90 | 2004 | 1002 |
+| 3010 | 1309.95 | 10/06/90 | 2004 | 1002 |
+| 3008 |    4723 | 10/05/90 | 2005 | 1001 |
+| 3011 | 9891.88 | 10/06/90 | 2006 | 1001 |
+| 3002 |  1900.1 | 10/03/90 | 2007 | 1004 |
+| 3001 |   18.69 | 10/03/90 | 2008 | 1007 |
+| 3006 | 1096.16 | 10/03/90 | 2008 | 1007 |
++------+---------+----------+------+------+
+10 rows in set (0.00 sec)
+
+
+5) Find which salespeople currently have orders in the order table. 
+
+mysql> SELECT salespeople.sname
+    -> FROM salespeople
+    -> LEFT JOIN orders
+    -> ON salespeople.snum = orders.snum
+    -> GROUP BY salespeople.sname;
++---------+
+| sname   |
++---------+
+| Peel    |
+| Serres  |
+| AxelRod |
+| Motika  |
+| Rifkin  |
+| Fran    |
++---------+
+6 rows in set (0.00 sec)
+
+
+
+6) List names of all customers matched with the salespeople serving them. 
+
+mysql> SELECT cname
+    -> FROM customers
+    -> LEFT JOIN salespeople ON customers.snum = salespeople.snum;
++----------+
+| cname    |
++----------+
+| Hoffman  |
+| Giovanni |
+| Liu      |
+| Grass    |
+| Clemens  |
+| Cisneros |
+| Pererira |
++----------+
+7 rows in set (0.00 sec)
+
+
+7) Find the names and numbers of all salespeople who have more than one customer. 
+
+mysql> SELECT sname, snum
+    ->     FROM salespeople
+    ->     WHERE snum IN ( SELECT snum
+    ->                     FROM customers
+    ->			   GROUP BY snum
+    ->                     HAVING COUNT(snum) > 1);
++--------+------+
+| sname  | snum |
++--------+------+
+| Peel   | 1001 |
+| Serres | 1002 |
++--------+------+
+2 rows in set (0.00 sec)
+
+
+8) Count the orders of each of the salespeople and output the results in descending order. 
```

Comment:
Edge case missed
Try to think about this
if there is salespeople who is not associated with any order
Expected
In that case the order count of salespeople shows as 0.